SU2_CFD/include/solver_structure.hpp

```diff
@@ -156,8 +166,29 @@ class CSolver {
   
   string SolverName;      /*!< \brief Store the name of the solver for output purposes. */
   
-  su2double valResidual;          /*!< \brief Store the residual of the linear system solution. */
-  
+  su2double valResidual;  /*!< \brief Store the residual of the linear system solution. */
+
+  /*!
+   * \brief Pure virtual function, all derived solvers MUST implement a method returning their "nodes".
+   * \note Don't forget to call SetBaseClassPointerToNodes() in the constructor of the derived CSolver.
+   * \return Nodes of the solver, upcast to their base class (CVariable).
+   */
+  virtual CVariable* GetBaseClassPointerToNodes() = 0;
+
+  /*!
+   * \brief Call this method to set "base_nodes" after the "nodes" variable of the derived solver is instantiated.
+   * \note One could set base_nodes directly if it were not private but that could lead to confusion
+   */
+  inline void SetBaseClassPointerToNodes() { base_nodes = GetBaseClassPointerToNodes(); }
+
+private:
+
+  /*--- Private to prevent use by derived solvers, each solver MUST have its own "nodes" member of the
+   most derived type possible, e.g. CEulerVariable has nodes of CEulerVariable* and not CVariable*.
+   This variable is to avoid two virtual functions calls per call i.e. CSolver::GetNodes() returns
+   directly instead of calling GetBaseClassPointerToNodes() or doing something equivalent. ---*/
+  CVariable* base_nodes;  /*!< \brief Pointer to CVariable to allow polymorphic access to solver nodes. */
+
```

Comment:
Ok, three votes in favour, so it is done.
It is all explained here in the comments, this was the cleanest way I could think of avoiding confusion by having two variables accessible in derived solvers that could be used seemingly interchangeably.
If a new CSolver is added people will invariably end up reading these comments about how to setup the system and runtime checks are performed for debug builds inside the GetNodes() method.